opacclient/libopac/src/main/java/de/geeksfactory/opacclient/apis/SLUB.kt

```diff
@@ -159,19 +159,26 @@ open class SLUB : OkHttpBaseApi() {
 
     override fun getResultById(id: String, homebranch: String?): DetailedItem {
         val json: JSONObject
+        val url =
+                if (id.startsWith("id")) {
+                    "$baseurl/$id/"
+                } else {
```

Comment:
Maybe you should also check here whether the ID starts with `bc/` and provide a fallback otherwise (e.g. for items saved on the favorites list before this change).

Reply:
Hm, old items from the favorites list seem to be the only possible case where you could encounter an id without a prefix. I think as the API is brand new no-one would have already used it with favorites.  
There are also `rsn` identifiers (not used yet) than can be resolved to `id` identifiers the same way with this code, so the correct and universal way to verify whether the identifier is an old un-prefixed `id` identifier would be by checking that neither the third nor the fourth character is a `/`. If you think we really need it I can add it, no problem - it's your choice.

Reply:
Hm, I would still try to be as explicit as possible (also because the code then better documents which possibilities there are for an ID) - so check for all cases, such as:
```kotlin
val url = when (id) {
    startsWith("id/") -> "$baseurl/$id/"
    startsWith("bc/"), startsWith("rsn/") -> // ...
    else -> "$baseurl/id/$id"
}
```

Reply:
done, see 360b9e5 (though the `when` didn't work out so nicely as suggested).

Reply:
Ah, right, I thought `when` would work like this, but apparently it doesn't. Still okay though :)